README.md

```diff
@@ -337,3 +337,128 @@ NAMESPACE_CLEANUP
 HTTP_STRESS_CONTAINER_IMAGE
 HTTP_SERVER_CONTAINER_IMAGE
 ```
+
+## RHCOS Network tests Workload
+
+The RHCOS network tests workload playbook is `workloads/network.yml` and is used to test latency and throughput of the cluster network.
+
+Requirements:
+* All Network tests - labeling nodes
+* HostNetwork - privileged pods
+* HostNetwork - Cloud firewall configuration to permit traffic for ports between 20010-20100 and NETWORK_TEST_UPERF_SSHD_PORT
+
+Running from CLI:
+
+```sh
+$ cp workloads/inventory.example inventory
+$ # Add orchestration host to inventory
+$ # Edit vars in workloads/vars/network.yml or define Environment vars (See below)
+$ time ansible-playbook -vv -i inventory workloads/network.yml
+```
+
+### Environment variables for RHCOS network workload playbook
+
+```
+###############################################################################
+# Ansible SSH variables.
+###############################################################################
+PUBLIC_KEY
+PRIVATE_KEY
+
+ORCHESTRATION_USER
+###############################################################################
+# RHCOS workload variables.
+###############################################################################
+WORKLOAD_IMAGE
+
+WORKLOAD_JOB_NODE_SELECTOR
+WORKLOAD_JOB_TAINT
+WORKLOAD_JOB_PRIVILEGED
+
+KUBECONFIG_FILE
+
+PBENCH_SSH_PRIVATE_KEY_FILE
+PBENCH_SSH_PUBLIC_KEY_FILE
+ENABLE_PBENCH_AGENTS
+PBENCH_SERVER
+
+SCALE_CI_RESULTS_TOKEN
+JOB_COMPLETION_POLL_ATTEMPTS
+
+# Network benchmarks specific parameters:
+NETWORK_TEST_UPERF_IMAGE
+NETWORK_TEST_UPERF_SSHD_PORT
+NETWORK_TEST_UPERF_PORT_RANGE
+NETWORK_TEST_PREFIX
+NETWORK_TEST_PAIRS
+NETWORK_TEST_HOSTNETWORK
+NETWORK_TEST_SERVICE
+NETWORK_TEST_TYPE
+NETWORK_TEST_RUNTIME
+NETWORK_TEST_MESSAGE_SIZES
+NETWORK_TEST_PROTOCOL
+NETWORK_TEST_INSTANCES
+NETWORK_TEST_SAMPLES
+NETWORK_TEST_CLEANUP
+```
+
+### Notes for Network testing execution
+
+#### Suggested Pod2Pod Test Configuration
+
+```
+NETWORK_TEST_UPERF_IMAGE="quay.io/openshift-scale/scale-ci-uperf"
+NETWORK_TEST_UPERF_SSHD_PORT=20000
+NETWORK_TEST_UPERF_PORT_RANGE="20010 20109"
+NETWORK_TEST_PREFIX=network
+NETWORK_TEST_PAIRS=1,2,4
+NETWORK_TEST_HOSTNETWORK=false
+NETWORK_TEST_SERVICE=false
+NETWORK_TEST_TYPE=stream,rr
+NETWORK_TEST_RUNTIME=60
+NETWORK_TEST_MESSAGE_SIZES=64,1024,16384
+NETWORK_TEST_PROTOCOL=tcp,udp
+NETWORK_TEST_INSTANCES=1,2
+NETWORK_TEST_SAMPLES=3
+NETWORK_TEST_CLEANUP=true
+```
+
+#### Suggested Pod2Pod HostNetwork Test Configuration
+
+```
+NETWORK_TEST_UPERF_IMAGE="quay.io/openshift-scale/scale-ci-uperf"
+NETWORK_TEST_UPERF_SSHD_PORT=20000
+NETWORK_TEST_UPERF_PORT_RANGE=""
+NETWORK_TEST_PREFIX=network
+NETWORK_TEST_PAIRS=1
```

Comment:
I know most of the other tests don't have this, but I don't understand what `NETWORK_TEST_PAIRS` is unless I look through the code.  Documentation of specific test variables would be great not only for this test, but all tests we have. In fact, documenting these vars should be compulsory.

Reply:
In this case pairs refers to client/server uperf pairs. (It gets confusing in that instances refers to how many uperf threads run).  Also I agree in the same sense I don't understand all tests's vars and what are the values for each.  I think this should play into issue #16 though.  The most immediate solution that comes to mind is a docs folder with index and page for each workload but I would like to get some ideas from the group before paving that road.